framework/OpenMod.Runtime/Runtime.cs

```diff
@@ -291,6 +295,35 @@ public async Task PerformSoftReloadAsync()
             await InitAsync(m_OpenModHostAssemblies!, m_RuntimeInitParameters!, m_HostBuilderFunc);
         }
 
+        private void PerformFileSystemWatcherPatch()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && Type.GetType("Mono.Runtime") != null)
```

Comment:
chances are high that the same issue exists in .NET we likely can remove mono check here

Reply:
https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System/services/io/system/io/FileSystemWatcher.cs#L979

If check the source code of .NET it use the kernel32 events instead of big thread in mono.